include/max_element.hpp

```diff
@@ -21,11 +21,54 @@ namespace bit {
 
 
 
-// Status: to do
+// Status: complete
 template <class ForwardIt>
 constexpr bit_iterator<ForwardIt> max_element(bit_iterator<ForwardIt> first,
     bit_iterator<ForwardIt> last) {
-    (last);
+
+    using word_type = typename bit_iterator<ForwardIt>::word_type;
+    constexpr std::size_t num_digits = bit::binary_digits<word_type>::value; 
+
+    bit_iterator<ForwardIt> cursor = first;
+
+    // handle misaligned first word. for example ^00100^101
+    if (!_is_aligned_lsb(cursor)) {
+        word_type first_word = _padded_read(cursor, last, bit::bit0);
+        if (first_word > 0) {
+            std::size_t first_set_bit_position = _tzcnt(first_word);
+            return cursor + (first_set_bit_position - cursor.position());
+        }
+
+        if (_in_same_word(cursor, last) ||
+            _is_one_past_last_bit(last, cursor.base())) {
+            return first;
+        }
+
+        std::advance(cursor, num_digits - cursor.position());
```

Comment:
If we use whole word iterators, this is another line that we could get rid of I think.